### PR TITLE
Release v3.0.0-preview

### PR DESCRIPTION
**Pull request by jiangxb1987**
Include 3.0.0-preview release in the download page and documentation.md, add news post to announce Spark 3.0.0 preview release.

Please note it intentionally didn't change the `latest` version, as the previous `2.0.0-preview` didn't  change it either.

### CODE REVIEW

**Comment by jiangxb1987** [comment]
cc @srowen @dongjoon-hyun @HyukjinKwon @gatorsmile @cloud-fan

File: news/_posts/2019-11-06-spark-3.0.0-preview.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+The Spark issue tracker already contains a list of <a href="https://issues.apache.org/jira/browse/SPARK-26078?jql=statusCategory%20%3D%20done%20AND%20project%20%3D%2012315420%20AND%20fixVersion%20%3D%2012339177%20ORDER%20BY%20priority%20DESC%2C%20key%20ASC">features in 3.0</a> and <a href="https://issues.apache.org/jira/browse/SPARK-27471?jql=project%20%3D%2012315420%20AND%20%22Target%20Version%2Fs%22%20%20%3D%2012339177%20AND%20statusCategory%20in%20(%22To%20Do%22%2C%20%22In%20Progress%22)%20ORDER%20BY%20priority%20DESC%2C%20key%20ASC">known issues in the preview</a>.
```
Comment (cloud-fan) [inline news/_posts/2019-11-06-spark-3.0.0-preview.md]: is this the right link?

File: js/downloads.js [hunk 2]
```diff
@@ -14,14 +14,18 @@ function addRelease(version, releaseDate, packages, mirrored) {
 var sources = {pretty: "Source Code", tag: "sources"};
 var hadoopFree = {pretty: "Pre-built with user-provided Apache Hadoop", tag: "without-hadoop"};
 var hadoop2p6 = {pretty: "Pre-built for Apache Hadoop 2.6", tag: "hadoop2.6"};
-var hadoop2p7 = {pretty: "Pre-built for Apache Hadoop 2.7 and later", tag: "hadoop2.7"};
+var hadoop2p7 = {pretty: "Pre-built for Apache Hadoop 2.7", tag: "hadoop2.7"};
+var hadoop3p2 = {pretty: "Pre-built for Apache Hadoop 3.2 and later", tag: "hadoop3.2"}
```
Comment (cloud-fan) [inline js/downloads.js:18]: shall we mention that this is built with JDK 11?

File: news/_posts/2019-11-06-spark-3.0.0-preview.md (hunk 1 above)
Comment (jiangxb1987) [reply news/_posts/2019-11-06-spark-3.0.0-preview.md]: Since there is no `3.0.0-preview` version in JIRA, I used version code for `3.0.0` instead.

Comment (cloud-fan) [reply news/_posts/2019-11-06-spark-3.0.0-preview.md]: When I open these 2 links, it's just a single ticket, not a list.

File: js/downloads.js (hunk 2 above)
Comment (srowen) [reply js/downloads.js:18]: It's not built with JDK 11; it still works on Java 8.

File: news/_posts/2019-11-06-spark-3.0.0-preview.md (hunk 1 above)
Comment (srowen) [reply news/_posts/2019-11-06-spark-3.0.0-preview.md]: The links work OK for me; I get a list.
I wouldn't include the second link; they're not necessarily known issues nor even things that will be 3.0. (We need to proactively prune that list but I'm not sure most people have.)

File: js/downloads.js (hunk 2 above)
Comment (cloud-fan) [reply js/downloads.js:18]: does it work with JDK 11?

Comment (srowen) [reply js/downloads.js:18]: Yes, that's the idea! hope so.

File: news/_posts/2019-11-06-spark-3.0.0-preview.md (hunk 1 above)
Comment (jiangxb1987) [reply news/_posts/2019-11-06-spark-3.0.0-preview.md]: Thanks, updated.

**Review by srowen — APPROVED** [review]

**Comment by jiangxb1987** [comment]
Thanks @srowen @cloud-fan !

**Comment by dongjoon-hyun** [comment]
Thank you, @jiangxb1987 . Sorry for being late.

**Pull request merged**